oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,262 @@
+.. _this pull request: https://github.com/openedx/edx-platform/pull/30567
+
+Rationale for migrating to the `Transifex GitHub App`_
+******************************************************
+
+* This is an upgrade of a system we use regularly, but do not want to have to maintain
+  regularly.
+* Upgrading from a bot (machine user) to an app/workflow is recommended by GitHub and
+  makes the translation process more open source. 
+* The `Transifex GitHub App`_ is developed and supported by Transifex
+* The `Transifex GitHub App`_ is very simple to configure and has many options. We can
+  set Transifex Projects to automatically upload/download translation files from a
+  repository once the translations are reviewed and accepted.
+* By using an app that is maintained by Transifex the organization, we reduce the
+  maintenance burden and are more future proof of changes they might make since they
+  maintain both the API and the `Transifex GitHub App`_.
+
+Rationale for consolidating translations files centrally
+********************************************************
+
+* Transifex only allows a one-to-one relationship between repositories and Transifex
+  Projects. Organizing all of the translation files into one repository and one Transifex
+  Project has a lower labor cost: projects are managed separately so we end up spending
+  less time tracking translation progress, and debugging translation issues when all
+  translation files are put in the same place. By decreasing the number of projects we
+  need to maintain, we can add more content like the MFE translations.
+* A repository that only contains text/binary files, and uses branches to separate
+  translations related to Open edX releases can make all interactions with translations
+  very quick and simple due to the ability to clone and sparse-checkout the branch of a
+  specific release and the directory (repository name) with translation files.
+
+Proposed Implementation
+***********************
+
+Move Translation Files to a New Repo
+====================================
+
+Translation files (of types .mo and .po) currently exist amongst the code/documentation
+they translate. We will move these translation files from being amongst the
+code/documentation to their own repo. For example, a translation file for the openedx
+repository `edx-platform`_ located at
+``edx-platform/conf/locale/en/LC_MESSAGES/django.po`` would be moved to the new
+repository with the name openedx-translations_ located at
+``openedx-translation/edx-platform/conf/locale/en/LC_MESSAGES/django.po``. For easier
+reintegration, translation files will be kept in the same directory structure as the
+code/documentation they translate.
+
+Repositories that generate translation files will have their translation files generated
+and committed via a pull request to the openedx-translation repository via a GitHub
+workflow. Once the translation files from edx-platform and other repositories are moved
```

Comment:
Its almost exactly like that. Checkout a run of the github action: https://github.com/openedx/openedx-translations/actions/runs/3415232508

That's a good point about translation changes made to release branches.
Though the above is just the current iteration of the action, in the future we can make the action run on the latest release branch as well to keep everything updated while it is supported.

Reply:
How will the release branch logistics work exactly? I was worried because the [Transifex docs](https://help.transifex.com/en/articles/6265125-github-via-transifex-ui#h_de29ef5cec) state that "the GitHub integration works on top of a single branch."

Reply:
This is a good topic for [Follow-up Work](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3578494977/Translations+Management+Design+Implementation). 

As a first stab: when a new release is cut, a new branch of the translations corresponding to the release branch will also be made. And then a new project can track that branch on Transifex. We can update the GitHub Action to run for multiple branches to keep the translation source files for the new release up to date. And with every new release, the old release's transifex project will be archived, but the translations will still be available on GitHub in the old release's branch. 

The first outstanding issue to this approach would be making sure that Translators have access to this new project as well. It will be up to the Translations WG to manage the latest translation files for the entire platform as well as minor fixes to the latest release.

Reply:
I am wildly guessing here as I'm not involved in BTR, but I had assumed current state is that translations are frozen at the time the release is cut, and any needed changes are made directly to the translation files in the release's branch for that repo.

(Not saying this is what we should do in the future, just wanted to make sure I am on the right page.)

It might also be nice to include a way for `openedx-atlas` to sideload translation files for the correct `openedx-translation` named release branch being deployed/developed on. Not sure how that would work. A `--branch` parameter? Some sort of auto-detection?

Reply:
> It might also be nice to include a way for `openedx-atlas` to sideload translation files for the correct `openedx-translation` named release branch being deployed/developed on. Not sure how that would work. A `--branch` parameter? Some sort of auto-detection?

the readme on https://github.com/openedx/openedx-atlas documents 2 ways to specify which branch to pull:
* using `--config` and specifying a branch in a config file
* directly specifying the branch using `-b`/`--branch`